File: app/src/main/java/com/github/android/lvrn/lvrnproject/service/form/TrashDependedForm.java
package com.github.android.lvrn.lvrnproject.service.form;

import androidx.annotation.NonNull;

import com.github.valhallalabs.laverna.persistent.entity.base.TrashDependedEntity;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public abstract class TrashDependedForm<T extends TrashDependedEntity> extends ProfileDependedForm<T> {

    protected boolean isTrash = false;

    public TrashDependedForm(@NonNull String profileId, boolean isTrash) {
        super(profileId);
        this.isTrash = isTrash;
    }

    public boolean isTrash() {
        return isTrash;
    }

    public void setTrash(boolean trash) {
        isTrash = trash;
    }
}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/service/form/TrashDependedForm.kt
package com.github.android.lvrn.lvrnproject.service.form

import com.github.valhallalabs.laverna.persistent.entity.base.TrashDependedEntity

abstract class TrashDependedForm<T : TrashDependedEntity>
    (profileId: String, protected open val isTrash: Boolean) : ProfileDependedForm<T>(profileId) {
}